### PR TITLE
BUG: Handle rolling to contract with auto-close date outside window

### PR DESCRIPTION
**Pull request by yankees714**
This fixes a bug with the edge case where:

- A contract (C_0) in the chain has an auto-close date equal to the `end` date passed to get_rolls
- The contract following C_0 (C_1), is *not* the active contract relative to the contract after it (C2), e.g. C_1 has lower volume than C_2, etc.

In this case, we need to roll from C_2 back to C_0, in which we would previously jump to the day following C_0's auto-close date. However, in this edge case, that would be outside the desired window (past `end`). Instead, in this case, we'll jump to `end` instead.

Also includes some cosmetic cleanup.

### CODE REVIEW

File: tests/test_continuous_futures.py [hunk 1]
```diff
@@ -1523,6 +1558,27 @@ def test_end_before_auto_close(self):
...
+            start=self.last_start_date,
+            end=self.fifth_auto_close_date,
+            offset=0,
+        )
+        self.assertEqual(
+            rolls,
+            [
+                (1003, pd.Timestamp('2017-04-18', tz='UTC')),
+                (1004, pd.Timestamp('2017-05-19', tz='UTC')),
+                (1006, None),
```
Comment (dmichalowicz) [inline tests/test_continuous_futures.py]: This is failing for me.
